### PR TITLE
enh(conf) add find host template by id

### PR DESCRIPTION
**Pull request by a-launois**
## Description

Add findHostTemplateById method in HostConfigurationService

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

File: src/Centreon/Infrastructure/HostConfiguration/Repository/HostConfigurationRepositoryRDB.php [hunk 1]
```diff
@@ -1065,4 +1065,34 @@ public function findHostByName(string $hostName): ?Host
...
+                host.host_alias AS alias,
+                host.host_register AS type,
+                host.host_activate AS is_activated
+            FROM `:db`.host
+            WHERE
+                host.host_id = :host_id AND
+                host.host_register = \'0\''
+        );
+        $statement = $this->db->prepare($request);
+        $statement->bindValue(':host_id', $hostTemplateId, \PDO::PARAM_STR);
```
Comment (kduret) [inline src/Centreon/Infrastructure/HostConfiguration/Repository/HostConfigurationRepositoryRDB.php:1087]:
```suggestion
        $statement->bindValue(':host_id', $hostTemplateId, \PDO::PARAM_INT);
```

File: src/Centreon/Infrastructure/HostConfiguration/Repository/HostConfigurationRepositoryRDB.php [hunk 2]
```diff
@@ -1065,4 +1065,34 @@ public function findHostByName(string $hostName): ?Host
...
+        $statement->execute();
+
+        if (($record = $statement->fetch(\PDO::FETCH_ASSOC)) !== false) {
+            return EntityCreator::createEntityByArray(
+                    Host::class,
+                    $record
+                );
```
Comment (kduret) [inline src/Centreon/Infrastructure/HostConfiguration/Repository/HostConfigurationRepositoryRDB.php:1094]:
```suggestion
                Host::class,
                $record
            );
```

File: src/Centreon/Domain/HostConfiguration/HostConfigurationService.php [hunk 3]
```diff
@@ -688,4 +688,17 @@ public function findHostByName(string $hostName): ?Host
             throw new HostConfigurationException(_('Error while searching for the host'), 0, $ex);
         }
     }
+
+    /**
+     * @inheritDoc
```
Comment (kduret) [inline src/Centreon/Domain/HostConfiguration/HostConfigurationService.php:693]:
```suggestion
     * {@inheritDoc}
```

File: src/Centreon/Domain/HostConfiguration/HostConfigurationService.php [hunk 4]
```diff
@@ -688,4 +688,17 @@ public function findHostByName(string $hostName): ?Host
             throw new HostConfigurationException(_('Error while searching for the host'), 0, $ex);
         }
     }
+
+    /**
+     * {@inheritDoc}
+     * @throws HostConfigurationException
+     */
+    public function findHostTemplateById(int $hostTemplateId): ?Host
```
Comment (jeremyjaouen) [inline src/Centreon/Domain/HostConfiguration/HostConfigurationService.php:696]:
```suggestion
    public function findHostTemplate(int $hostTemplateId): ?Host
```
While not precising "By..." it is implicitly ById

File: src/Centreon/Domain/HostConfiguration/Interfaces/HostConfigurationReadRepositoryInterface.php [hunk 5]
```diff
@@ -135,4 +135,13 @@ public function findHostTemplatesByHost(Host $host): array;
...
     public function findHostByName(string $hostName): ?Host;
+
+    /**
+     * Find host template by its id
+     *
+     * @param int $hostTemplateId
+     * @return Host|null
+     * @throws \Throwable
+     */
+    public function findHostTemplateById(int $hostTemplateId): ?Host;
```
Comment (jeremyjaouen) [inline src/Centreon/Domain/HostConfiguration/Interfaces/HostConfigurationReadRepositoryInterface.php:146]:
```suggestion
    public function findHostTemplate(int $hostTemplateId): ?Host;
```

File: src/Centreon/Domain/HostConfiguration/Interfaces/HostConfigurationServiceInterface.php [hunk 6]
```diff
@@ -140,4 +140,12 @@ public function findHostTemplatesByHost(Host $host): array;
...
      */
     public function findHostByName(string $hostName): ?Host;
+
+    /**
+     * Find a host template by its id
+     *
+     * @param int $hostTemplateId
+     * @return Host|null
+     */
+    public function findHostTemplateById(int $hostTemplateId): ?Host;
```
Comment (jeremyjaouen) [inline src/Centreon/Domain/HostConfiguration/Interfaces/HostConfigurationServiceInterface.php:150]:
```suggestion
    public function findHostTemplate(int $hostTemplateId): ?Host;
```

File: src/Centreon/Infrastructure/HostConfiguration/Repository/HostConfigurationRepositoryRDB.php [hunk 7]
```diff
@@ -1065,4 +1065,34 @@ public function findHostByName(string $hostName): ?Host
         }
         return null;
     }
+
+    /**
+     * @inheritDoc
+     */
+    public function findHostTemplateById(int $hostTemplateId): ?Host
```
Comment (jeremyjaouen) [inline src/Centreon/Infrastructure/HostConfiguration/Repository/HostConfigurationRepositoryRDB.php:1072]:
```suggestion
    public function findHostTemplate(int $hostTemplateId): ?Host
```

**Review by jeremyjaouen — CHANGES_REQUESTED** [review]

**Review by jeremyjaouen — DISMISSED** [review]

**Review by kduret — DISMISSED** [review]

**Review by kduret — DISMISSED** [review]

File: src/Centreon/Domain/HostConfiguration/HostConfigurationService.php [hunk 8]
```diff
@@ -688,4 +688,24 @@ public function findHostByName(string $hostName): ?Host
...
+    {
+        try {
+            return $this->hostConfigurationRepository->findHostTemplate($hostTemplateId);
+        } catch (\Throwable $ex) {
+            $this->error(
+                sprintf(
+                    _('Error while searching for the host template'),
+                    $hostTemplateId,
+                    $ex->getMessage()
+                )
```
Comment (jeremyjaouen) [inline src/Centreon/Domain/HostConfiguration/HostConfigurationService.php:706]:
```suggestion
                sprintf(
                    _('Error while searching for the host template %d'),
                    $hostTemplateId
                ),
                [$ex->getMessage()]
```

File: lang/es_ES.UTF-8/LC_MESSAGES/messages.po [hunk 9]
```diff
@@ -15103,3 +15103,6 @@ msgstr ""
 
 # msgid "Do not use configured proxy to connect to this server"
 # msgstr ""
+
+#  msgid "Error while searching for the host template"
```
Comment (jeremyjaouen) [inline lang/es_ES.UTF-8/LC_MESSAGES/messages.po:15107]:
```suggestion
#  msgid "Error while searching for the host template %d"
```

**Review by jeremyjaouen — CHANGES_REQUESTED** [review]

**Review by callapa — APPROVED** [review]

**Review by jeremyjaouen — APPROVED** [review]

**Pull request merged**